### PR TITLE
Prevent read group set requests with...

### PR DESCRIPTION
**Pull request by dcolligan**
...nonexistant names from issuing 404s

Issue #647

### CODE REVIEW

File: ga4gh/backend.py [hunk 1]
```diff
@@ -389,6 +389,13 @@ def _singleObjectGenerator(self, datamodelObject):
         """
         yield (datamodelObject.toProtocolElement(), None)
 
+    def _noObjectGenerator(self):
+        """
+        Returns a generator yielding no results
+        """
+        return
```
Comment (jeromekelleher) [inline ga4gh/backend.py]: This looks a bit weird --- how about 
```
return iter([])
```

**Comment by jeromekelleher** [comment]
Looks good @dcolligan, but we really should add some test cases to cover this. ``test_simulated_stack.py`` is probably the best place to do this. In fact, the existing test ``testCallSetsSearch`` should be failing now, as it's asserting the old 404 behaviour; we should get to the bottom of this.

**Comment by dcolligan** [comment]
@jeromekelleher updated.  I don't see why testCallSetsSearch would be affected.  This change only affects the path for searching for read group sets.

**Comment by jeromekelleher** [comment]
Ah yes, my bad. We do need to fix CallSets too, so we shouldn't close #647 until that's done.

+1.

**Pull request merged**